From 6b4dc8aad4a5208ce0f147faa53beede196064fb Mon Sep 17 00:00:00 2001
From: Cody Peter Mello <cody.mello@joyent.com>
Date: Wed, 23 May 2018 16:34:15 +0000
Subject: [PATCH] NAPI-434 Want way to automatically generate available subnets
 (fix unit tests) Reviewed by: Nick Zivkovic <nick.zivkovic@joyent.com>
 Approved by: Nick Zivkovic <nick.zivkovic@joyent.com>

---
 test/integration/fabrics.test.js           | 10 ++++++++--
 test/unit/migration-1.2.0-networks.test.js |  3 ++-
 2 files changed, 10 insertions(+), 3 deletions(-)

diff --git a/test/integration/fabrics.test.js b/test/integration/fabrics.test.js
index 040bff0..e07d9c6 100644
--- a/test/integration/fabrics.test.js
+++ b/test/integration/fabrics.test.js
@@ -37,7 +37,7 @@ var extend = mod_jsprim.mergeObjects;
 // --- Globals
 
 
-var client = h.createNAPIclient();
+var client;
 
 var subnet_alloc_enabled;
 var ADMIN_OWNER = config.server.ufdsAdminUuid;
@@ -265,10 +265,16 @@ function checkEventLog(t, opts) {
 
 
 test('setup', function (t) {
+    client = h.createNAPIclient();
+
     t.test('create default nic tag', mod_nic_tag.createDefault);
     t.test('is-alloc-enabled', function (t2) {
         client.ping(function (err, res) {
-            t2.ok(err === null);
+            if (h.ifErr(t2, err, 'ping() error')) {
+                t2.end();
+                return;
+            }
+
             subnet_alloc_enabled = res.config.subnet_alloc_enabled;
             t2.end();
         });
diff --git a/test/unit/migration-1.2.0-networks.test.js b/test/unit/migration-1.2.0-networks.test.js
index 44263b0..2a4e531 100644
--- a/test/unit/migration-1.2.0-networks.test.js
+++ b/test/unit/migration-1.2.0-networks.test.js
@@ -576,7 +576,8 @@ test('setup', function (t) {
             t2.ifError(err, 'ping() error');
             t2.deepEqual(status, {
                 config: {
-                    fabrics_enabled: true
+                    fabrics_enabled: true,
+                    subnet_alloc_enabled: false
                 },
                 healthy: true,
                 services: {
-- 
2.21.0

